test/test_nn.py

```diff
@@ -20001,6 +20163,64 @@ def perm_fn(x):
             self.assertEqual(result.shape, ref_output.shape)
             torch.testing.assert_close(result, ref_output, atol=atol, rtol=rtol)
 
+            # NestedTensor is only supported for the fast path
+            # currently, which won't be used if training.
+            if batch_first and not training and ('cuda' in str(device) or 'cpu' in str(device)):
+                encoder_input[0][-1] = torch.zeros_like(encoder_input[0][1])
+                mask = torch.zeros(encoder_input.shape[:-1], device=device, dtype=torch.bool)
+                mask[0][-1] = True
+
+                nt = torch.nested_tensor([encoder_input[0][:-1], encoder_input[1]], device=device)
+                result = model(nt)
+                ref_output = torch.tensor(
```

Comment:
Couldn't we call the slow path to get these reference instead of hard-coding them with "random" accuracy?